src/V3Timing.cpp

```diff
@@ -12,11 +12,27 @@
 // Version 2.0.
 // SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
 //
-// TimingSuspendableVisitor locates all C++ functions and processes that contain timing controls,
-// and marks them as suspendable. If a process calls a suspendable function, then it is also marked
-// as suspendable. If a function calls or overrides a suspendable function, it is also marked as
-// suspendable. TimingSuspendableVisitor creates a dependency graph to propagate this property. It
-// does not perform any AST transformations.
+// TimingSuspendableVisitor does not perform any AST transformations.
+// Instead it propagates two types of flags:
+// - flag "suspendable": (for detecting what will need to become a coroutine)
+//     The visitor locates all C++ functions and processes that contain timing controls,
+//     and marks them as suspendable. If a process calls a suspendable function,
+//     then it is also marked as suspendable. If a function calls or overrides
+//     a suspendable function, it is also marked as suspendable.
+//     TimingSuspendableVisitor creates a dependency graph to propagate this property.
+// - flag "needs process": (for detecting what needs a VlProcess argument in signature)
+//   The visitor distinguishes 4 types of nodes:
+//     1. nodes that can allocate VlProcess (forks, always, initial etc.),
+//     2. nodes that make it necessary for the previous type to allocate VlProcess
+//        (like process::self which then wraps it, allowing use inside Verilog).
+//     3. nodes that should obtain VlProcess if it will be allocated
+//        (all of the previous type + timing controls, so they could update process state),
+//     4. nodes that are going to be emitted with a VlProcess argument.
+//   flags 2. and 3. are propagated upwards up to the nodes type 1., this is to detect
```

Comment:
```suggestion
//   flags 2. and 3. are propagated upwards up to the nodes of type 1., this is to detect
```
Also I think it'd be more readable to spell out the flag names.

Reply:
Done.